scripts/jenkins/qat/Makefile

```diff
@@ -0,0 +1,26 @@
+# Copyright 2019 Intel Corporation.
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+# Makefile for QAT-DPDK jenkins.
+
+qat: qat-checks qat-pull qat-plugin qat-tests
+
+qat-checks:
+	bash checks.sh
+
+qat-pull:
+	bash images-pull.sh
```

Comment:
the toplevel `Makefile` could have `pull` target that pulls all relevant images. The script that handles pulling must be generic and accept a list of images (with the registry/tag information) to pull.

Reply:
As we discussed previously, push can be something easy cause only docker is used to do so, root makefile itself assumes that talking abuut push, but in the case of pull, well there are more options, `ctr`, `crictl`, `docker`, we can specify the BUILDER="any" to execute the proper command, but  each one its own way to set credentials when pulling from a private registry, that is our case since we are pulling always a custom tag from a private registry that requires auth, and that is the only use case when we need to pull. I don't have a good ideas yet to handle the credentials in a generic way without using `if` or `case` for each container runtime. do you have something in mind?

Reply:
`crictl` is a safe choice for both cri-o and containerd so don't worry about the others (`ctr` is gone, `docker` we don't use).

The image pulling overall is problematic which I what I tried to explain yesterday. If we have a test cluster with several worker nodes, the images potentially need to be pulled to all of them.  Can you provision the worker images with keys that can access the registry?

Reply:
The same challenge is with `checks.sh`. Which worker node it checks?

Reply:
in a multinode cluster, running a single command for everything... maybe we will need to spend a lot of effort thinking and designing a good solution for that, sincerely i see no bad in splitting a little bit for each device in that case, since we are building something that can do that: e.g. in a single node cluster `make tests`,  in a multinode `make qat`, `make fpga`... bla bla bla..., I've been thinking In a jenkinsfile that can be seen like:

```
def devices = ["qat", "fpga", "..."]

def parallelDevicesStagesMap(job) {
  return {
    node("${job}") {
     stage("${job}") {
        sh 'make -C scripts/jenkins/ ${job}'
      }
    }
  }
}

Stage {
  agent { control-plane }
  ....
  stages {
    stage ("Tests") {
      steps {
        script {
          parallel parallelDevicesStagesMap
        }
      }
    }
  }
}```

Reply:
@chuyd  can we provision the nodes with keys, tokens or whatever they need to access to registry and pull/push withot specifying creeds?? (in a secure way of course)